File: FrontEnd/src/model/schedule.ts
import { Student } from "./student";
import { Tutor } from "./tutor";

export interface Schedule {
    id?: number;
    tutorId?: number;
    tutor: Tutor;
    name?: string;
    subjects?: Subject[];
    hourlyPrice?: number;
    isPaid?:boolean;
    startTime: Date;
    endTime: Date;
    studentId?: number;
    student: Student;
  }

  export enum Subject {
    Economics = "Economics",
    Maths = "Maths",
    Programming = "Programming",
    Startup = "Startup",
    PE = "PE",
    DiscMaths = "DiscMaths",
  }

  export interface State {
    Schedules: Schedule[];
  }

File: BackEnd/Model/Schedule.cs
using System.ComponentModel.DataAnnotations.Schema;
using tuutoriplatvorm.Model;

namespace backend.Model
{
    public class Schedule 
    {
        public int? Id { get; init; }
        public required int TutorId { get; init; }
        public Tutor Tutor { get; set; } // = null!;
        public int? StudentId { get; set; }
        public Student? Student { get; set; } //= null!;

      //  public string? Name { get; init; }
        public required List<Subject> Subjects { get; init; }
        public double? HourlyPrice { get; init; }

        public bool IsPaid { get; set; }
        public DateTime StartTime { get; init; }
        public DateTime EndTime { get; init; }

    }
}